cmd/eib/main.go

```diff
@@ -1,7 +1,74 @@
 package main
 
-import "fmt"
+import (
+	"flag"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/suse-edge/edge-image-builder/pkg/config"
+)
+
+const (
+	argConfigFile = "config"
+	argVerbose    = "verbose"
+)
+
+func init() {
+	log.SetFormatter(&log.TextFormatter{
+		FullTimestamp:    true,
+		QuoteEmptyFields: true,
+	})
+	log.SetOutput(os.Stdout)
+}
+
+func processArgs() (imageConfig *config.ImageConfig) {
+	var (
+		configFile string
+		verbose    bool
+	)
+
+	flag.StringVar(&configFile, argConfigFile, "", "full path to the image configuration file")
+	flag.BoolVar(&verbose, argVerbose, false, "enables extra logging information")
+	flag.Parse()
+
+	handleVerbose(verbose)
+	imageConfig = handleImageConfig(configFile)
+
+	return imageConfig
+}
+
+func handleVerbose(verbose bool) {
+	if verbose {
+		log.SetLevel(log.DebugLevel)
+	}
+}
+
+func handleImageConfig(configFile string) *config.ImageConfig {
```

Comment:
It'd be better for this function to return (`*config.ImageConfig, error`), evaluate the error when calling it from `main()` and if present - exit the app with a non-0 code (i.e. `log.Fatalf()`).

The key takeaway is to avoid having unexpected parts of the code causing an early exit. Functions like this one should either return the expected value on success or error on failure - it's not different from the perfectly designed [Parse()](https://github.com/suse-edge/edge-image-builder/blob/main/pkg/config/image.go#L14) one you added previously.

Reply:
Makes sense, not sure why I changed the strategy between that one and this.